exec/cnex/lib/binary.c

```diff
@@ -0,0 +1,478 @@
+    range_check32(exec, y);
+    push(exec->stack, cell_fromNumber(number_from_uint32(number_to_uint32(x) ^ number_to_uint32(y))));
+}
+
+void binary_and64(TExecutor *exec)
+{
+    Number y = top(exec->stack)->number; pop(exec->stack);
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    range_check64(exec, x);
+    range_check64(exec, y);
+    push(exec->stack, cell_fromNumber(number_from_uint64(number_to_uint64(x) & number_to_uint64(y))));
+}
+
+void binary_extract64(TExecutor *exec)
+{
+    Number w = top(exec->stack)->number; pop(exec->stack);
+    Number n = top(exec->stack)->number; pop(exec->stack);
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    range_check64(exec, x);
+
+    range_checkU64(exec, n);
+    range_checkU64(exec, w);
+    unsigned int b = number_to_uint32(n);
+    if (b >= 64) {
+        push(exec->stack, cell_fromNumber(BID_ZERO));
+        return;
+    }
+    unsigned int v = number_to_uint32(w);
+    if (b + v > 64) {
+        v = 64 - b;
+    }
+    if (v == 64) {
+        push(exec->stack, cell_fromNumber(x));
+        return;
+    }
+    push(exec->stack, cell_fromNumber(number_from_uint64((number_to_uint64(x) >> b) & (1 << v) - 1)));
+}
+
+void binary_get64(TExecutor *exec)
+{
+    Number n = top(exec->stack)->number; pop(exec->stack);
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    range_check64(exec, x);
+    range_checkU64(exec, n);
+    unsigned int b = number_to_uint32(n);
+    if (b >= 64) {
+        push(exec->stack, cell_fromBoolean(FALSE));
+        return;
+    }
+    push(exec->stack, cell_fromBoolean((number_to_uint64(x) & 1ULL << b) != 0));
+}
+
+void binary_not64(TExecutor *exec)
+{
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    range_check64(exec, x);
+    push(exec->stack, cell_fromNumber(number_from_uint64(~number_to_uint64(x))));
+}
+
+void binary_or64(TExecutor *exec)
+{
+    Number y = top(exec->stack)->number; pop(exec->stack);
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    range_check64(exec, x);
+    range_check64(exec, y);
+    push(exec->stack, cell_fromNumber(number_from_uint64(number_to_uint64(x) | number_to_uint64(y))));
+}
+
+void binary_replace64(TExecutor *exec)
+{
+    Number y = top(exec->stack)->number; pop(exec->stack);
+    Number w = top(exec->stack)->number; pop(exec->stack);
+    Number n = top(exec->stack)->number; pop(exec->stack);
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    range_check64(exec, x);
+    range_checkU64(exec, n);
+    range_checkU64(exec, w);
+    range_check64(exec, y);
+    unsigned int b = number_to_uint32(n);
+    if (b >= 64) {
+        push(exec->stack, cell_fromNumber(BID_ZERO));
+        return;
+    }
+    unsigned int v = number_to_uint32(w);
+    if (b + v > 64) {
+        v = 64 - b;
+    }
+    uint64_t z      = number_to_uint64(y);
+    uint64_t mask   = v < 64 ? (number_to_uint64(BID_ONE) << v) - number_to_uint64(BID_ONE) : ~0;
```

Comment:
More instances of `number_to_uint64(BID_ONE)`. Make sure you get all of these.

Reply:
Fixed.  Removed all instances, and replaced with literal 1.